src/modm/driver/rtc/mcp7941_impl.hpp

```diff
@@ -25,69 +25,38 @@ template < typename I2cMaster >
 modm::mcp7941::Data&
 modm::Mcp7941<I2cMaster>::getData()
 {
-	return reinterpret_cast<Data&>(this->data);
+	return this->data;
 }
 
 template < typename I2cMaster >
-inline modm::ResumableResult<modm::mcp7941::DateTime>
+inline modm::ResumableResult<bool>
 modm::Mcp7941<I2cMaster>::getDateTime(){
 	RF_BEGIN();
-	this->transaction.configureWriteRead(&addr_seconds, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.seconds = data.data[0];
-	this->transaction.configureWriteRead(&addr_minutes, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.minutes = data.data[0];
-	this->transaction.configureWriteRead(&addr_hours, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.hours = data.data[0];
-	this->transaction.configureWriteRead(&addr_days, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.days = data.data[0];
-	this->transaction.configureWriteRead(&addr_months, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.months = data.data[0];
-	this->transaction.configureWriteRead(&addr_years, 1, data.data, 1);
-	RF_CALL(this->runTransaction());
-	data.date_time.years = data.data[0];
-
-	RF_END_RETURN(data.date_time);
+	this->transaction.configureWriteRead(&addr_seconds, 1, scratch, 7);
+	if(RF_CALL(this->runTransaction())){
+		data.date_time.seconds = scratch[0];
+		data.date_time.seconds = scratch[1];
+		data.date_time.minutes = scratch[2];
+		data.date_time.hours = scratch[3];
+		data.date_time.days = scratch[4];
+		data.date_time.months = scratch[5];
+		data.date_time.years = scratch[6];	
+	}
+	RF_END_RETURN(true);
```

Comment:
You probably want to return the result of `runTransaction()` from L36.
Easiest way is to add `RF_RETURN(true)` to the end of the if block and an else block returning false. (With plain `RF_END()` at the end of the function.)

Reply:
agreed, will fix, thanks